File: java/src/main/java/exams/exams1/Exam1.java
package exams.exams1;

/**
 * 1-两数之和
 *
 * @author ijlhjj
 * @version 1.0 2023-05-28
 */
public class Exam1 {

    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2]; //存放结果

        for (int i = 0; i < nums.length - 1; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    result[0] = i;
                    result[1] = j;
                    return result;
                }
            }
        }

        return result;
    }

}

File: kotlin/src/main/kotlin/exams/exams1/Exam1.kt
package exams.exams1

/**
 * 使用 Map 辅助，达到时间复杂度：O(N)
 */
fun twoSum(nums: IntArray, target: Int): IntArray {
    var indexMap = mutableMapOf<Int, Int>()

    for ((index, value) in nums.withIndex()) {
        if ((target - value) in indexMap) return intArrayOf(index, indexMap.getValue(target - value))

        indexMap[value] = index
    }

    return intArrayOf()
}
